src/coreclr/src/vm/readytoruninfo.h

```diff
@@ -48,13 +75,18 @@ class ReadyToRunInfo
 
     PTR_PersistentInlineTrackingMapR2R m_pPersistentInlineTrackingMap;
 
-    ReadyToRunInfo(Module * pModule, PEImageLayout * pLayout, READYTORUN_HEADER * pHeader, AllocMemTracker *pamTracker);
-
 public:
+    ReadyToRunInfo(Module * pModule, PEImageLayout * pLayout, READYTORUN_HEADER * pHeader, NativeImage * pNativeImage, AllocMemTracker *pamTracker);
+
     static BOOL IsReadyToRunEnabled();
 
     static PTR_ReadyToRunInfo Initialize(Module * pModule, AllocMemTracker *pamTracker);
 
+    bool IsComponentAssembly() const { return m_isComponentAssembly; }
+    const ReadyToRunCoreInfo * GetComponentInfo() const { return &m_component; }
+    const ReadyToRunCoreInfo * GetCompositeInfo() const { return m_pComposite; }
+    NativeImage * GetNativeImage() const { return m_pNativeImage; }
```

Comment:
The vast majority of the uses of this function are GetCompositeInfo()->GetImage(). I feel that there should be just a GetImage() function on this type instead to reduce the verbosity, and hold the invariant that ReadyToRunInfo can only come from one file.

Also, given that that leaves only 2-3 uses of GetCompositeInfo() left, I think it might be worth doing the refactoring to entirely hide away the concept of composite vs component from consumers of this class.

Reply:
I love this suggestion, fixed in 11th commit.